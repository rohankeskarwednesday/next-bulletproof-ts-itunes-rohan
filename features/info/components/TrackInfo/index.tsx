import { If, T } from "@common";
import { Card, Tag } from "antd";
import isEmpty from "lodash/isEmpty";
import { ReactComponentLike } from "prop-types";
import React, { memo } from "react";
import { injectIntl, IntlShape } from "react-intl";
import { compose } from "redux";
import Link from "next/link";

const { Meta } = Card;

interface TrackInfoProps {
  intl: IntlShape;
  data: any;
}

const TrackInfo: React.FC<TrackInfoProps> = ({ intl, data }) => {
  const trackInfo = data.results[0];

  const {
    trackCensoredName,
    collectionCensoredName,
    artistName,
    primaryGenreName,
    artworkUrl100,
    kind,
    trackViewUrl,
    collectionViewUrl,
  } = trackInfo;

  const colors = [
    "magenta",
    "red",
    "volcano",
    "orange",
    "gold",
    "lime",
    "green",
    "cyan",
    "blue",
    "geekblue",
    "purple",
  ];

  const BlockText = props => <T display="block" {...props} />;

  return (
    <div
      style={{
        width: "100%",
        height: "100vh",
        display: "flex",
        justifyContent: "center",
      }}
    >
      <Card
        style={{ width: 400, height: "auto", maxHeight: 630, margin: "20px" }}
        cover={
          <div
            style={{
              backgroundImage: `url(${artworkUrl100.replace("100x100", "600x600")})`,
              width: "100%",
              height: "400px",
              backgroundPosition: "center",
              backgroundSize: "cover",
              backgroundRepeat: "no-repeat",
            }}
          ></div>
        }
      >
        <Meta
          title={trackCensoredName ? trackCensoredName : collectionCensoredName}
          description={artistName}
        />
        <If condition={!isEmpty(primaryGenreName)}>
          <p style={{ marginTop: "20px" }} data-testid="track-genre">
            {intl.formatMessage({ id: "track_genre" })}
            <Tag
              style={{ borderRadius: "100px", marginLeft: "10px" }}
              color={colors[Math.floor(Math.random() * (colors.length - 0 + 1)) + 0]}
            >
              {primaryGenreName}
            </Tag>
          </p>
        </If>
        <If condition={!isEmpty(kind)}>
          <BlockText
            id="track_kind"
            data-testid="track-kind"
            values={{ kind: kind }}
            style={{ fontSize: "15px", marginTop: "10px" }}
          />
        </If>
        <If condition={!isEmpty(collectionViewUrl)}>
          <Link href={collectionViewUrl}>
            <a href={collectionViewUrl} target="__blank" data-testid="view-details">
              <p style={{ marginTop: "10px", cursor: "pointer", color: "blue" }}>
                {intl.formatMessage({ id: "view_track_details" })}
              </p>
            </a>
          </Link>
        </If>
      </Card>
    </div>
  );
};

export default compose(injectIntl, memo)(TrackInfo) as ReactComponentLike;
