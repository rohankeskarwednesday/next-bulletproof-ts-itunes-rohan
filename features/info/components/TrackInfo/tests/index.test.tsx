import { render } from "@utils/testUtils";
import TrackInfo from "../index";

describe("<TrackInfo />", () => {
  const songInfoProps = {
    data: {
      results: [
        {
          wrapperType: "track",
          kind: "song",
          artistId: 420203509,
          collectionId: 1440858222,
          trackId: 1440858321,
          artistName: "Bastille",
          collectionName: "Bad Blood (Bonus Track Version)",
          trackName: "Pompeii",
          collectionCensoredName: "Bad Blood (Bonus Track Version)",
          trackCensoredName: "Pompeii",
          artistViewUrl: "https://music.apple.com/us/artist/bastille/420203509?uo=4",
          collectionViewUrl:
            "https://music.apple.com/us/album/pompeii/1440858222?i=1440858321&uo=4",
          trackViewUrl: "https://music.apple.com/us/album/pompeii/1440858222?i=1440858321&uo=4",
          previewUrl:
            "https://audio-ssl.itunes.apple.com/itunes-assets/AudioPreview115/v4/1e/e3/20/1ee32037-f1a3-e46b-011f-76552e7c5fa2/mzaf_6001471003681391544.plus.aac.p.m4a",
          artworkUrl30:
            "https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/e6/d8/61/e6d86177-ae8c-e84e-dfcc-0042687066ed/13UAAIM41955.rgb.jpg/30x30bb.jpg",
          artworkUrl60:
            "https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/e6/d8/61/e6d86177-ae8c-e84e-dfcc-0042687066ed/13UAAIM41955.rgb.jpg/60x60bb.jpg",
          artworkUrl100:
            "https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/e6/d8/61/e6d86177-ae8c-e84e-dfcc-0042687066ed/13UAAIM41955.rgb.jpg/100x100bb.jpg",
          collectionPrice: 6.99,
          trackPrice: 1.29,
          releaseDate: "2013-01-11T12:00:00Z",
          collectionExplicitness: "notExplicit",
          trackExplicitness: "notExplicit",
          discCount: 1,
          discNumber: 1,
          trackCount: 15,
          trackNumber: 1,
          trackTimeMillis: 214148,
          country: "USA",
          currency: "USD",
          //   primaryGenreName: "Alternative",
          isStreamable: true,
        },
      ],
    },
  };

  it("should not show track genre", () => {
    const { queryByTestId } = render(<TrackInfo {...songInfoProps} />);

    const genreName = queryByTestId("track-genre");

    expect(genreName).toBe(null);
  });

  it("should show view details link", () => {
    const { queryByTestId } = render(<TrackInfo {...songInfoProps} />);

    const genreName = queryByTestId("view-details");

    expect(genreName).not.toBe(null);
  });
});
