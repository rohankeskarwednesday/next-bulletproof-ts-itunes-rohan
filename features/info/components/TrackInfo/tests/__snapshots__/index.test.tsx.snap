// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackInfo /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      style="width: 100%; height: 100vh; display: flex; justify-content: center;"
    >
      <div
        class="ant-card ant-card-bordered"
        style="width: 400px; height: 400px; margin: 20px;"
      >
        <div
          class="ant-card-cover"
        >
          <div
            style="background-image: url(https://is1-ssl.mzstatic.com/image/thumb/Music115/v4/e6/d8/61/e6d86177-ae8c-e84e-dfcc-0042687066ed/13UAAIM41955.rgb.jpg/600x600bb.jpg); width: 100%; height: 200px; background-position: center; background-size: cover; background-repeat: no-repeat;"
          />
        </div>
        <div
          class="ant-card-body"
        >
          <div
            class="ant-card-meta"
          >
            <div
              class="ant-card-meta-detail"
            >
              <div
                class="ant-card-meta-title"
              >
                Pompeii
              </div>
              <div
                class="ant-card-meta-description"
              >
                Bastille
              </div>
            </div>
          </div>
          <p
            data-testid="track-genre"
            style="margin-top: 20px;"
          >
            Genre:
            <span
              class="ant-tag ant-tag-cyan"
              style="border-radius: 100px; margin-left: 10px;"
            >
              Alternative
            </span>
          </p>
          <p
            data-testid="track-kind"
            style="margin-top: 10px; font-size: 12px;"
          >
            <p
              class="T__StyledText-sc-apdcd0-0 fjJkvM"
              data-testid="t"
              display="block"
              style="font-size: 15px;"
            >
              Kind: song
            </p>
          </p>
          <a
            href="https://music.apple.com/us/album/pompeii/1440858222?i=1440858321&uo=4"
            target="__blank"
          >
            <p
              style="margin-top: 10px; cursor: pointer; color: blue;"
            >
              View details
            </p>
          </a>
        </div>
      </div>
    </div>
  </div>
</body>
`;
